File: javascript/src/POSDisplayer.js
const { DetailFrame } = require('./DetailFrame');
/** @typedef {import('./ServletRequest').ServletRequest} ServletRequest */

class POSDisplayer {
  constructor() {
    this._pageStateName = 'stage1';
    this._detailDisplay = undefined;
    this._marketBindings = {};
    this._displayDirection = [0, 0, 0, 0];
  }
  /**
   *
   * @param {ServletRequest} request
   */
  populate(request) {
    const values = request.getParameterValues(this._pageStateName);

    if (values != null && values.length > 0) {
      this._marketBindings[this._pageStateName + this._getDateStamp()] =
        values[0];
    }
    const detailDisplay = new DetailFrame();
    detailDisplay.setDescription(
      this._getDetailText() + ' ' + this._getProjectionText()
    );
    detailDisplay.show();
    this._setDisplayRegion(
      0,
      0,
      detailDisplay.frameLength,
      detailDisplay.frameWidth
    );
  }
  _setDisplayRegion(aX, aY, bX, bY) {
    this._displayDirection[0] = aX;
    this._displayDirection[1] = aY;
    this._displayDirection[2] = bX;
    this._displayDirection[3] = bY;
  }
  showMenu() {
    return this._displayDirection;
  }
  _getDetailText() {
    return 'detail text';
  }
  _getProjectionText() {
    return 'projection info';
  }
  _getDateStamp() {
    return new Date().toString();
  }
}

module.exports = { POSDisplayer };

File: java/src/main/java/POSDisplayer.java
import java.awt.TextField;
import java.awt.event.ActionEvent;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;


public class POSDisplayer {

    private String pageStateName;
    private DetailFrame detailDisplay;
    private Map marketBindings;
    private int[] displayDirection = new int[4];

    public POSDisplayer() {
        marketBindings = new HashMap<String,String>();
        pageStateName = "stage1";

    }

    public void process(ServletRequest request) {
        String [] values
                = request.getParameterValues(pageStateName);
        if (values != null && values.length > 0) {
            marketBindings.put(pageStateName + getDateStamp(),
                               values[0]);
        }

        detailDisplay = new DetailFrame();
        detailDisplay.setDescription(
                    getDetailText() + " " + getProjectionText());
        detailDisplay.show();
        setDisplayRegion(0,0,detailDisplay.getFrameLength(),detailDisplay.getFrameWidth());
    }

    private void setDisplayRegion(int aX, int aY, int bX, int bY) {
        displayDirection[0] = aX;
        displayDirection[1] = aY;
        displayDirection[2] = bX;
        displayDirection[3] = bY;
    }

    public int[] getMenuRegion() {
        return displayDirection;
    }

    private String getDetailText() {
        return "detail text";
    }

    private String getProjectionText() {
        return "projection info";
    }

    private String getDateStamp() {
        return new Date().toString();
    }
}
